SocketRocket/SRWebSocket.m

```diff
@@ -1467,8 +1501,48 @@ - (void)stream:(NSStream *)aStream handleEvent:(NSStreamEvent)eventCode;
             }
                 
             case NSStreamEventHasSpaceAvailable: {
-                SRFastLog(@"NSStreamEventHasSpaceAvailable %@", aStream);
-                [self _pumpWriting];
+                if (_secure) {
+                    // First parameter determines the certificate type (yes=server, no=everything else)
+                    // Second parameter specifies whether to validate the hostname (setup in the certificates CNAME) or to ignore it if set to NULL.
+                    SecPolicyRef policy = SecPolicyCreateSSL(NO, NULL);
+                    SecTrustRef trust = NULL;
+                    
+                    // get the locally available peer certificates
+                    CFArrayRef cert = (__bridge CFArrayRef)[aStream propertyForKey:(NSString *) kCFStreamPropertySSLPeerCertificates];
+                    // create the SecTrust
+                    SecTrustCreateWithCertificates(cert, policy, &trust);
+                    // create the trust anchor
+                    
+                    SecTrustSetAnchorCertificates(trust, (__bridge CFArrayRef) [_urlRequest SR_SSLPinnedCertificates]);
+                    
+                    SecTrustResultType trustresult;
+                    // main function to evaluate all settings and certificate information
+                    OSStatus evalStatus = SecTrustEvaluate(trust, &trustresult);
+                    if (evalStatus == errSecSuccess) {
+                        // kSecTrustResultUnspecified defines the use of self signed certificates or in other words the missing user specified settings concering the trust
+                        // in this case proceed as normal
+                        if (trustresult == kSecTrustResultUnspecified && self.allowSelfSignedCertificates) {
+                            NSLog(@"Trusted certificate. Result: %u", trustresult);
```

Comment:
Please don't use `NSLog` to report errors, otherwise all of SocketRocket's other customers are going to see these messages -- in production even -- and not know why! :)